nibabel/spatialimages.py

```diff
@@ -325,6 +326,21 @@ class SpatialImage(DataobjImage):
     ''' Template class for volumetric (3D/4D) images '''
     header_class = SpatialHeader
 
+    class Slicer(object):
+        ''' Slicing interface that returns a new image with an updated affine
+        '''
+        def __init__(self, img):
+            self.img = img
+
+        def __getitem__(self, slicer):
+            try:
+                slicer = self.img._check_slicing(slicer)
+            except ValueError as err:
+                raise IndexError(*err.args)
+            dataobj = self.img.dataobj[slicer]
+            affine = self.img._slice_affine(slicer)
```

Comment:
It's a bit scary using this apparently private method.  Did we discuss this before?  I forget?

Reply:
https://github.com/nipy/nibabel/pull/550#discussion_r141847919